programs/adapter-lido/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program::invoke,
+    pubkey::Pubkey,
+};
+use anchor_spl::token::{Mint, TokenAccount};
+
+declare_id!("ADPTPxbHbEBo9A8E53P2PZnmw3ZYJuwc8ArQQkbJtqhx");
+
+#[program]
+pub mod adapter_lido {
+    use anchor_lang::solana_program::{account_info::next_account_infos, stake};
+
+    use super::*;
+
+    pub fn deposit<'a, 'b, 'c, 'info>(
+        ctx: Context<'a, 'b, 'c, 'info, Action<'info>>,
+        input: Vec<u8>,
+    ) -> Result<()> {
+        // Get Input
+        let mut input_bytes = &input[..];
+        let input_struct = DepositInputWrapper::deserialize(&mut input_bytes)?;
+        
+        msg!("Input: {:?}", input_struct);
+     
+        let deposit_accounts = vec![
+            AccountMeta::new(ctx.remaining_accounts[0].key(), false),
+            AccountMeta::new(ctx.remaining_accounts[1].key(), true),
+            AccountMeta::new(ctx.remaining_accounts[2].key(), false),
+            AccountMeta::new(ctx.remaining_accounts[3].key(), false),
+            AccountMeta::new(ctx.remaining_accounts[4].key(), false),
+            AccountMeta::new_readonly(ctx.remaining_accounts[5].key(), false),
+            AccountMeta::new_readonly(ctx.remaining_accounts[6].key(), false),
+            AccountMeta::new_readonly(ctx.remaining_accounts[7].key(), false),  
+        ];
+
+        let ix = Instruction { program_id: ctx.accounts.base_program_id.key(), accounts: deposit_accounts, data: input };
+        
+        invoke(&ix, ctx.remaining_accounts)?;
+
+        Ok(())
+    }
+
+    pub fn withdraw<'a, 'b, 'c, 'info>(
+        ctx: Context<'a, 'b, 'c, 'info, Action<'info>>,
+        input: Vec<u8>,
+    ) -> Result<()> {
+        // Get Input
+        let mut input_bytes = &input[..];
+        let input_struct = WithdrawInputWrapper::deserialize(&mut input_bytes)?;
+
+        msg!("Input: {:?}", input_struct);
+        
+        let withdraw_accounts = if input_struct.instruction == 23 {
```

Comment:
Actually we can distinguish v1 or v2 with the remaining account length, so instruction can write in adapter without passing in.

Reply:
Thanks, I modified it do this now